src/main/java/net/openhft/chronicle/wire/converter/SizeLongConverter.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2016-2020 chronicle.software
+ *
+ *       https://chronicle.software
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.openhft.chronicle.wire.converter;
+
+import net.openhft.chronicle.wire.LongConverter;
+
+/**
+ * A converter for long values that translates between numeric values and
+ * their string representations with a size suffix (K, M, G, T) to indicate
+ * the magnitude (kilo, mega, giga, tera). This class implements the
+ * {@link LongConverter} interface, providing methods to parse a string
+ * with a size suffix back into a long value and to convert a long value
+ * into its string representation with the appropriate size suffix.
+ */
+public class SizeLongConverter implements LongConverter {
+    public static final SizeLongConverter INSTANCE = new SizeLongConverter();
+
+    /**
+     * Parses a string representation of a number with a size suffix
+     * (K, M, G, T) and converts it into a long value. The suffix indicates
+     * the magnitude (kilo for K, mega for M, giga for G, tera for T) of the
+     * number, multiplying the parsed number by 1024 to the power corresponding
+     * to the suffix.
+     *
+     * @param text The text to parse, containing a numeric value followed by
+     *             a size suffix (K, M, G, T). Case-insensitive.
+     * @return The parsed long value, adjusted for the size suffix. If no
+     * suffix is provided, returns the plain numeric value.
+     * @throws NumberFormatException If the text before the size suffix is not
+     *                               a valid long number or if the text is empty.
+     */
+    @Override
+    public long parse(CharSequence text) throws NumberFormatException {
+        int shift = 0;
```

Comment:
Can consider throwing NPE with custom message if `text==null`

Reply:
it will on the next line.